test/com/minseungkwon/datastructure/Graph/BinaryTreeTest.java

```diff
@@ -116,4 +116,80 @@ public void test_BSTtoList() {
 
         assertEquals("4/17/0258/369/", bt.BSTtoList());
     }
+
+    @Test
+    public void test_isBalancedUsingTwoRecursiveTrueCase() {
+        Integer[] arr = new Integer[10];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = i;
+        }
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+
+        assertEquals(true, bt.isBalancedUsingTwoRecursive());
+    }
+
+    @Test
+    public void test_isBalancedUsingTwoRecursiveFalseCase() {
+        Integer[] arr = new Integer[10];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = i;
+        }
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+        bt.getRoot().right.right.right.right = new Node(10);
+
+        assertEquals(false, bt.isBalancedUsingTwoRecursive());
+    }
+
+    @Test
+    public void test_isBalancedUsingOneRecursiveTrueCase() {
+        Integer[] arr = new Integer[10];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = i;
+        }
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+
+        assertEquals(true, bt.isBalancedUsingOneRecursive());
+    }
+
+    @Test
+    public void test_isBalancedUsingOneRecursiveFalseCase() {
+        Integer[] arr = new Integer[10];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = i;
+        }
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+        bt.getRoot().right.right.right.right = new Node(10);
+
+        assertEquals(false, bt.isBalancedUsingOneRecursive());
+    }
+
+    @Test
+    public void test_isBalancedUsingObjectTrueCase() {
+        Integer[] arr = new Integer[10];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = i;
+        }
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+
+        assertEquals(true, bt.isBalancedUsingObject());
+    }
+
+    @Test
+    public void test_isBalancedUsingObjectFalseCase() {
+        Integer[] arr = new Integer[10];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = i;
+        }
+        BinaryTree<Integer> bt = new BinaryTree<>();
+        bt.makeTree(arr);
+        bt.getRoot().right.right.right.right = new Node(10);
+        bt.getRoot().right.right.left = new Node(11);
```

Comment:
Why this test case has 11th node?

Reply:
Since this one uses different logic from the other two methods. In this case, its result is false, but the results for the other two methods are true. I added the same cases for the two methods that result in true.